docs: update documentation for RDT monitoring/metrics. (#549)
containers/nri-plugins: docs/resource-policy/policy/common-functionality.md

This is not possible with the NRI Reference Plugins configuration CR. Here you
must use the latter full syntax.

## Cache Occupancy Monitoring Metrics

> Plugins can be configured to export cache usage prometheus metrics data collected
> by goresctrl. To enable these metrics, make sure that in addition to having RDT
> classes/CLOSes set up in your configuration, you also have

Reviewer: Is there actually a need to specify any classes in the RDT configuration? Can't it be run in a "discovery" mode to use whatever was configured outside the plugin?

```suggestion
Plugins can be configured to export cache usage as Prometheus metrics. The following configuration options must be specified:
```

Author: > Is there actually a need to specify any classes in the RDT configuration? Can't it be run in a "discovery" mode to use whatever was configured outside the plugin?

@marquiz Currently if any class assignments are to be done by the plugin then classes need to be specified (otherwise we'll end up with no resctrl CLOSes/classes, so creating any container assigned to a class will fail in the runtime). But, PTAL #551 which enables running in goresctrl in discovery mode.

  - `control.rdt.enable` set to `true`
  - `instrumentation.prometheusExport` set to `true`,
  - `instrumentation.httpEndpoint` set to a valid non-empty value, eg. `:8891`, and
  - `instrumentation.metrics.enabled` set to contain `policy/rdt`, `rdt`, or `policy`

> With helm you can modify the default configuration accordingly like this:

Reviewer: nit: improve wording (dunno if this is really that much better, though...)

```suggestion
When deploying with Helm, the default configuration can be modified like this:
```

```shell
helm install test -n kube-system nri-plugins/nri-resource-policy-topology-aware \
    --set config.control.rdt.enable=true \
    --set config.instrumentation.prometheusExport=true \
    --set config.instrumentation.metrics.enabled='{buildinfo,rdt}' \
    --set config.log.debug='{goresctrl}'
```

> Once you enable RDT metrics collection, you should get per CLOS metrics like this:

Reviewer: nit: wording, more generic (should we mention monitoring group metrics here?)

```suggestion
Once enabled, you’ll see RDT metrics metrics similar to the following:
```

```shell

> kubectl port-forward -n kube-system ds/nri-resource-policy-topology-aware 9000:8891 &
> wget -q --no-proxy http://127.0.0.1:9000/metrics -O-

Reviewer: ```suggestion
$ kubectl port-forward -n kube-system ds/nri-resource-policy-topology-aware 9000:8891 &
wget -q --no-proxy http://127.0.0.1:9000/metrics -O-
```

Author: Updated.

          mbAllocation:
            all: [ 100%, 1000Mbps ]
          classes:
            BestEffort:
              l2Allocation:
                all:
                  unified: 33%
              l3Allocation:
                all:
                  unified: 33%
              mbAllocation:
                all: [ 33%, 330Mbps ]
            Burstable:
              l2Allocation:
                all:
                  unified: 66%
              l3Allocation:
                all:
                  unified: 66%
              mbAllocation:
                all: [ 66%, 660Mbps ]
            Guaranteed:
              l2Allocation:
                all:
                  unified: 100%
              l3Allocation:
                all:
                  unified: 100%
              mbAllocation:
                all: [ 100%, 1000Mbps ]
```

## Memory Bandwidth Monitoring Metrics

If you have RDT-specific metrics collection enabled and your platform supports
memory bandwidth monitoring, you can expect these related metrics to be exposed:

  - l3_mbm_local_bytes: bytes transferred to/from local memory through LLC
  - l3_mbm_total_bytes: total bytes transferred to/from memory through LLC

> with extra data like this:

Reviewer: Hmm, strange sentence, maybe e.g.

```suggestion
An example:
```

```shell

> kubectl port-forward -n kube-system ds/nri-resource-policy-topology-aware 9000:8891 &

Reviewer: nit:

```suggestion
$ kubectl port-forward -n kube-system ds/nri-resource-policy-topology-aware 9000:8891 &
```
